tests/test_copy_novaseqx_flow_cell.py

```diff
@@ -0,0 +1,126 @@
+    )
+
+    # THEN it is not ready
+    assert not ready
+
+
+def test_flow_cell_is_not_demultiplexed(
+    demultiplex_not_complete_novaseqx_flow_cell: Path, demultiplexed_flow_cell_run_directory: Path
+):
+    # GIVEN a flow cell for which demultiplexing is not completed
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(
+        demultiplex_not_complete_novaseqx_flow_cell, demultiplexed_flow_cell_run_directory
+    )
+
+    # THEN it is not ready
+    assert not ready
+
+
+def test_previously_post_processed_flow_cell_is_not_ready(
+    post_processed_novaseqx_flow_cell: Path, demultiplexed_flow_cell_run_directory: Path
+):
+    # GIVEN a flow cell for which post processing is done
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(
+        post_processed_novaseqx_flow_cell, demultiplexed_flow_cell_run_directory
+    )
+
+    # THEN the flow cell is not ready
+    assert not ready
+
+
+def test_previously_copied_flow_cell_is_not_ready(
+    novaseqx_flow_cell_dir: Path, demultiplexed_flow_cell_run_directory: Path
+):
+    # GIVEN a flow cell which already exists in demultiplexed runs
+    Path(demultiplexed_flow_cell_run_directory, novaseqx_flow_cell_dir.name).mkdir()
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(
+        novaseqx_flow_cell_dir, demultiplexed_flow_cell_run_directory
+    )
+
+    # THEN the flow cell is not ready
+    assert not ready
+
+
+def test_get_latest_analysis_version_path(
+    novaseqx_flow_cell_dir: Path,
+    novaseqx_latest_analysis_version: str,
+):
+    # GIVEN a flow cell which is ready to be post processed
+
+    # WHEN extracting the latest analysis version path
+    analysis: Path = get_latest_analysis_directory(novaseqx_flow_cell_dir)
+
+    # THEN the latest analysis version path is returned
+    latest_analysis = Path(
+        novaseqx_flow_cell_dir,
+        DemultiplexingDirsAndFiles.ANALYSIS,
+        novaseqx_latest_analysis_version,
+    )
+    assert analysis == latest_analysis
+
+
+def test_copy_novaseqx_flow_cell(
+    demultiplexed_flow_cell_run_directory: Path,
+    novaseqx_flow_cell_dir: Path,
+    novaseqx_flow_cell_dir_name: str,
+):
+    # GIVEN a destination directory
+    flow_cell_run = Path(demultiplexed_flow_cell_run_directory, novaseqx_flow_cell_dir_name)
+    flow_cell_run.mkdir()
+    destination = Path(flow_cell_run, DemultiplexingDirsAndFiles.DATA)
+
+    # WHEN copying the flow cell analysis data to demultiplexed runs
+    copy_flow_cell_analysis_data(novaseqx_flow_cell_dir, destination)
+
+    # THEN the data contains everything from the analysis folder
+    analysis: Path = get_latest_analysis_directory(novaseqx_flow_cell_dir)
+    analysis_data = analysis / DemultiplexingDirsAndFiles.DATA
+
+    original_files = get_all_files(analysis_data)
+    copied_files = get_all_files(destination)
```

Comment:
```suggestion
    original_files: Set[Path] = get_all_files(analysis_data)
    copied_files: Set[Path] = get_all_files(destination)
```
It explains what we are asserting making it easier to understand the test.

Reply:
fixed